EIPS/eip-1288.md

```diff
@@ -0,0 +1,66 @@
+---
+eip: 1288
+title: Get contract return values from transaction recepts
+author: Aritz Beobide-Cardinal <aritz@aritzcracker.ca> 
+discussions-to: https://github.com/ethereum/EIPs/issues/1288
+status: Draft
+type: Standards Track
+category: Interface
+created: 2018-08-01
+---
+
+## Simple Summary
+This is a proposal to give clients the ability to easily get contract return values and revert reasons from already completed transactions.
+
+## Abstract
+This EIP proposes to allow eth_getTransactionReceipt to show return values from a smart contract executon, similarly to eth_call.
+
+## Motivation
+Clients have no reliable way to get _why_ a transaction has failed when mined, nor can they get the return value from a successful smart contract execution. A workaround would be to use eth_call before or after the transaction is mined, but that is unreliable since the state of a smart contract can change multiple times within the same block.
+
+For example, any eth_call checks or gas estimations may return OK before the transaction is submitted to the mempool, but your transaction may be reverted due to the actions of one that got in just before you.
+
+Expanding on this example, your transaction may have been between 2 other transactions which affected yours, so any eth_call checks on that block or the previous block return an unexpected value, resulting in much frustration. and confusion.
+
+## Specification
+
+Take the result of eth_call and put it in eth_getTransactionReceipt's returning object; using a field called `contractResult`.
```

Comment:
The JSON API isn't currently standardised via EIPs. That doesn't make this EIP automatically invalid, but note that client implementers may not pay any attention to an EIP trying to define a change to an API, and being accepted does not mean it will be implemented.

Also, this change would break light clients.